docs/content/doc/features/localization.en-us.md

```diff
@@ -31,3 +31,70 @@ After a translation has been accepted, it will be reflected in the main reposito
+```
+
+Only the `form` is provided to this template - not the operand value itself. This is to allow autogeneration of the forms at time of loading.
+
+These will be compiled to various different keys based on the available formats in the
+language. e.g. assuming the above key is in the English locale it becomes:
+
+```ini
+form.reach_limit_of_creation_plural_one = You have already reached your limit of %d repository.
+form.reach_limit_of_creation_plural_other = You have already reached your limit of %d repositories.
+
+```
+
+If the template format is too cumbersome forms can be directly created and they will
+be used in preference to the template generated variants.
+
+These keys should be used with the `.TrPlural` function. (Ordinals can be handled with `.TrOrdinal`.)
+
+Each language has different numbers of plural forms, in English (and a number of other
+languages) there are two plural forms:
+
+* `One`: This matches the singular form.
+* `Other`: This matches the plural form.
+
+Other languages, e.g. Mandarin, have no plural forms, and others many more.
+
+The possible forms are:
+
+* `Other` - the most common form and will often match to standard plural form.
+* `Zero` - matches a zeroth form, which in Latvian would match the form used for 10-20, 30 and so on.
+* `One` - matches the singular form in English, but in Latvian matches the form used for 1, 21, 31 and so on.
+* `Two` - matches the dual form used in for example Arabic for 2 items, but also more complexly in Celtic languages.
+* `Few` - matches the form used in Arabic for 3-10, 103-110, and the ternary form in Celtic languages. In Russian and Ukranian for 2-4, 22-24.
+* `Many` - matches the form used for large numbers in romance lanaguages like French, e.g. 1 000 000 *de* chat*s*, but in Russian and Ukranian it handles 0, 5~19, 100, 1000 and so on.
+
+Some plural forms are only relevant if the object being counted is of a certain
+grammatical gender or in certain tenses. Write your translation template appropriately to take account of this using `not` or `and` as appropriately.
+
+Translators may want to review the CLDR information for their language or look at
+`modules/translation/i18n/plurals/generate/plurals.xml`.
+
+Ordinal forms, e.g. 1st, 2nd, 3rd and so on can be handled with `.TrOrdinal`. These
+have the same forms as the plural forms, and we will use `_ordinal` as a base suffix
+in future.
+
+If the provided key does not have the `_plural` suffix or its format template is invalid, the specific per plural forms must be provided explicitly. The plural key fallback hierarchy is as follows:
+
+1. `${key}_${form}` in the locale.
+2. `${key}_other` in the locale.
+3. `${key}` in the locale.
+4. `${key}_${form}` in the default locale.
+5. `${key}_other` in the default locale.
+6. `${key}` in the default locale.
+7. Use the `${key}_${form}` directly as the format.
```

Comment:
Does it work with current Gitea's translation system?

For example, the keys are only for English, other languages do not have the change to have other keys?

Reply:
And some more thoughts: as a translator, I could only translate what I have seen. Even if I know Gitea's code, when I do translation, I do not want to add any new keys (personally), it's too difficult for me ... I doubt whether there are translators could benefit from this mechanism.

Reply:
I would not expect that we would want to provide ANY keys like this. The point is that the `_plural` and `_ordinal` suffix causes the auto-creation of these for you and therefore this should be definitely preferred.

Explaining the hierarchy is helpful though for people who want to write their own locale files.

Reply:
I've adjusted the wording here to make it clear that this really is TECHNICAL INFORMATION